src/main/java/teammates/logic/core/StudentsLogic.java

```diff
@@ -286,27 +284,35 @@ private String getSectionInvalidityInfo(List<StudentAttributes> mergedList) {
             if (currentStudent.section.equals(previousStudent.section)) {
                 studentsCount++;
             } else {
-                if (studentsCount > SECTION_SIZE_LIMIT) {
+                if (studentsCount > Const.StudentsLogicConst.SECTION_SIZE_LIMIT) {
                     invalidSectionList.add(previousStudent.section);
                 }
                 studentsCount = 1;
             }
 
-            if (i == mergedList.size() - 1 && studentsCount > SECTION_SIZE_LIMIT) {
+            if (i == mergedList.size() - 1 && studentsCount > Const.StudentsLogicConst.SECTION_SIZE_LIMIT) {
                 invalidSectionList.add(currentStudent.section);
             }
         }
 
-        StringBuilder errorMessage = new StringBuilder();
+        StringJoiner errorMessage = new StringJoiner(" ");
         for (String section : invalidSectionList) {
-            errorMessage.append(String.format(Const.StatusMessages.SECTION_QUOTA_EXCEED, section));
+            errorMessage.add(String.format(
+                    Const.StudentsLogicConst.ERROR_ENROLL_EXCEED_SECTION_LIMIT,
+                    Const.StudentsLogicConst.SECTION_SIZE_LIMIT, section));
+        }
+
+        if (!invalidSectionList.isEmpty()) {
+            errorMessage.add(String.format(
+                    Const.StudentsLogicConst.ERROR_ENROLL_EXCEED_SECTION_LIMIT_INSTRUCTION,
+                    Const.StudentsLogicConst.SECTION_SIZE_LIMIT));
         }
 
         return errorMessage.toString();
     }
 
     private String getTeamInvalidityInfo(List<StudentAttributes> mergedList) {
-
+        StringJoiner errorMessage = new StringJoiner(" ");
         StudentAttributes.sortByTeamName(mergedList);
 
         List<String> invalidTeamList = new ArrayList<>();
```

Comment:
It seems that using HashMap will improve the performance, but I am OK with it now.